packages/background/src/controllers/ExchangeRatesController.ts

```diff
@@ -93,7 +90,14 @@ export class ExchangeRatesController extends BaseController<ExchangeRatesControl
                     });
 
                     this.updateNetworkNativeCurrencyId(network);
-                    await this.updateExchangeRates();
+
+                    //Still use interval manager to avoid multiple request but execute immeditately.
+                    this._exchangeRateFetchIntervalController.tick(
+                        0,
+                        async () => {
+                            await this.updateExchangeRates();
```

Comment:
can we close previous connections before starting new ones? This will save resources on the browser and our nodes too

Reply:
`_exchangeRateFetchIntervalController` is preventing this code to run twice. Connections are closed by the http client 's AbortController